onboard/catkin_ws/src/controls/scripts/desired_state.py

```diff
@@ -72,7 +81,31 @@ def _on_power_received(self, power):
         Args:
             power: ROS Twist message corresponding to desired powers in local reference frame
         """
-        self.power = utils.parse_twist(power)
+        parsed_power = utils.parse_twist(power)
+        if self.power_state_safety(parsed_power):
+            self.power = parsed_power
+
+    def power_state_safety(self, power):
+        # Compares power with controller limits
+        return_status = True
+        for axes in utils.get_axes():
+            if power[axes] > self.maxpower[axes]:
```

Comment:
Should probably get the absolute value of the number, as negative speeds would not get caught here.

`if abs(power[axes]) > self.maxpower[axes]:`